internal/repository/ticket.go

```diff
@@ -19,6 +19,28 @@ func NewTicketRepository(db *gorm.DB) TicketRepository {
 	return TicketRepository{db: db}
 }
 
+func (repo *TicketRepository) BookTicket(ctx context.Context, ticket domain.Ticket) (domain.Ticket, error) {
+	if err := repo.db.WithContext(ctx).Create(&ticket).Error; err != nil {
+		return domain.Ticket{}, err
+	}
+	return ticket, nil
+}
+
+func (repo *TicketRepository) CheckAvailability(ctx context.Context, flightID uuid.UUID, totalTickets int) (bool, error) {
+	var count int64
+	if err := repo.db.WithContext(ctx).
```

Comment:
this query will be heavily used while booking. Think of optimizations that could be done. (hint: table itself)